File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {

        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode firstNode = dummy;
        ListNode secondNode = dummy;

        for(int i=0;i<n;i++){
            secondNode = secondNode.next;
        }

        while(secondNode.next != null){
            secondNode = secondNode.next;
            firstNode = firstNode.next;
        }
        firstNode.next = firstNode.next.next;

        return dummy.next;


    }
}

File: Data Structures & Algorithms/reverse-nodes-in-k-group/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        temp = head
        count = 0
        while count < k:
            if temp == None:
                return head
            temp = temp.next
            count+=1
        
        prevNode = self.reverseKGroup(temp,k);

        temp = head
        count =0;


        while count < k:
            nxt = temp.next
            temp.next = prevNode
            prevNode = temp
            temp = nxt
            count+=1
        
        return prevNode
        


        